### PR TITLE
Improve the code and add more details to the doc

### PR DESCRIPTION
**Pull request by hacksparrow**
Closes https://github.com/strongloop/loopback4-example-shopping/issues/69

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +20,65 @@ npm i
...
+
+This app has five models:
+
+1. `User` - representing the users of the system
+2. `Product` - a model which is mapped to a remote service by `services/recommender.service`
+3. `ShoppingCartItem` - a model for representing purchases
+4. `ShoppingCart` - a model to represent a user's shopping cart, can contain many items (`item`) of the type `ShoppingCartItem`
+5. `Order` - a model to represent an order by user, can have many products (`products`) of the type `ShoppingCartItem`
+
+`ShoppingCart` and `Order` are marked as belonging to the `User` model by the use of the `@belongsTo` model decorator. Correspondingly, the `User` model is marked as having many `Order`s using the `@hasMany` model decorator. Although possible, a `hasMany` relation for `User` for `ShoppingCart` has not be created in this particular app to limit the scope of the example.
```
Comment (b-admike) [inline README.md]: typo: `for 'User' for 'ShoppingCart' -> `for 'User' and 'ShoppingCart'

File: README.md [hunk 2]
```diff
@@ -20,6 +20,65 @@ npm i
...
+This app is intended to be interacted with using the API Explorer located at http://[::1]:3000/explorer/.
+
+## Models
+
+This app has five models:
+
+1. `User` - representing the users of the system.
+2. `Product` - a model which is mapped to a remote service by `services/recommender.service`.
+3. `ShoppingCartItem` - a model for representing purchases.
+4. `ShoppingCart` - a model to represent a user's shopping cart, can contain many items (`item`) of the type `ShoppingCartItem`.
```
Comment (jannyHou) [inline README.md]: `item` --> `items`?

File: README.md [hunk 3]
```diff
@@ -20,6 +20,65 @@ npm i
...
+4. `user-order` - controller for creating, updating, deleting orders, and getting the details about an order.
+
+## Services
+
+Services are modular components that can be plugged into a LoopBack application in various locations to contribute
+additional capabilities and features to the application.
+
+This app has five services:
+
+1. `services/hash.password.bcryptjs` - responsible for generating and comparing password hashes.
```
Comment (jannyHou) [inline README.md]: Nitpick: I would suggest organizing the auth related services together

1. `services/recommender.service` - responsible for connecting to a "remote" server and getting recommendations for a user. The API endpoint at `GET /users​/{userId}​/recommend`, is made possible by this service.
2. `services/user-service` - responsible for verifying if user exists and the submitted password matches that of the existing user.
3. `services/hash.password.bcryptjs` - responsible for generating and comparing password hashes.
4. `services/validator` - responsible for validating email and password when a new user is created.
5. `services/jwt-service` - responsible for generating and verifying JSON Web Token.

File: README.md [hunk 4]
```diff
@@ -20,6 +20,65 @@ npm i
...
+2. `services/jwt-service` - responsible for generating and verifying JSON Web Token.
+3. `services/recommender.service` - responsible for connecting to a "remote" server and getting recommendations for a user. The API endpoint at `GET /users​/{userId}​/recommend`, is made possible by this service.
+4. `services/user-service` - responsible for verifying if user exists and the submitted password matches that of the existing user.
+5. `services/validator` - responsible for validating email and password when a new user is created.
+
+## Login
+
```
Comment (jannyHou) [inline README.md]: *Note: This app contains a login endpoint for the purpose of spike and demo, the authentication for the CRUD operations and navigational endpoints of model `User` is still in build.*

File: README.md [hunk 5]
```diff
@@ -20,6 +20,65 @@ npm i
...
+The endpoint for logging in a user is a `POST` request to `/users/login`.
+
+Once the credentials are extracted, the logging-in implementation at the controller level is just a four step process. This level of simplicity is made possible by the use of the `UserService` service provided by
```
Comment (jannyHou) [inline README.md]: Good abstraction for the verification logic in `login`, we have an auth tutorial showing how the JWT strategy works in https://loopback.io/doc/en/lb4/Authentication-Tutorial.html which you can refer to :)

File: README.md [hunk 6]
```diff
@@ -20,6 +20,73 @@ npm i
...
+ `@loopback/authentication`.
+
+1. `const user = await this.userService.verifyCredentials(credentials)` - verify the credentials.
+2. `const userProfile = this.userService.convertToUserProfile(user)` - generate user profile object.
+3. `const token = await this.jwtService.generateToken(userProfile)` - generate JWT based on the user profile object.
+4. `return {token}` - send the JWT.
+
+You can see the details in `packages/shopping/src/controllers/user.controller.ts`.
```
Comment (nabdelgadir) [inline README.md]: Should we add a link to `packages/shopping/src/controllers/user.controller.ts`?

File: README.md [hunk 7]
```diff
@@ -59,13 +59,17 @@ additional capabilities and features to the application.
...
+1. `services/recommender.service` - responsible for connecting to a "remote" server and getting recommendations for a user. The API endpoint at `GET /users​/{userId}​/recommend`, is made possible by this service.
+2. `services/user-service` - responsible for verifying if user exists and the submitted password matches that of the existing user.
+3. `services/hash.password.bcryptjs` - responsible for generating and comparing password hashes.
+4. `services/validator` - responsible for validating email and password when a new user is created.
+5. `services/jwt-service` - responsible for generating and verifying JSON Web Token.
 
-## Login
+## Authentication
+
+*Note: This app contains a `login` endpoint for the purpose of spike and demo, the authentication for the CRUD operations and navigational endpoints of model User is still in build.*
```
Comment (b-admike) [inline README.md]: nitpick: `still in build.` -> `still in progress.`?

**Pull request merged**